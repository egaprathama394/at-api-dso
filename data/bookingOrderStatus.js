export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
