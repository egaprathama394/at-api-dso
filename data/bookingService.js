const randomstring = require('randomstring');

const fullNameMax = randomstring.generate({
  charset: 'string',
  length: '110',
});

const noPhoneMax = randomstring.generate({
  charset: 'numeric',
  length: '17',
});

const noPhoneRandom = randomstring.generate({
  charset: 'numeric',
  length: '18',
});

const fullNameRandom = randomstring.generate({
  charset: 'string',
  length: '111',
});

const randomNopol1 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol2 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol3 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol4 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

export default {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AB${randomNopol1}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght

export const maxInput = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameMax}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data booking service dengan inputan max char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneMax}`,
  policeRegNo: `AC${randomNopol2}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght + 1

export const negativeRequest = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameRandom}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service dengan inputan max + 1 char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneRandom}`,
  policeRegNo: `AD${randomNopol3}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for cancel booking service

export const inputDataForCancel = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data untuk cancel booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AA${randomNopol4}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const cancelRequest = {
  orderNo: '',
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// data for display booking service

export const displayBookingService = {
  isWalkIn: false,
  orderBy: '',
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const invalidDisplayBookingService = {
  isWalkIn: '',
  orderBy: '',
  AccountType: '',
  Action: '',
  ApplicationName: '',
  DeviceType: '',
  OperatingSystem: '',
  UserVersion: '',
};
