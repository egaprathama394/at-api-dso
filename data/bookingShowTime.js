export default {
  branchCode: 'D004',
  scheduleDate: '2020-09-30 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  branchCode: 'D004',
  scheduleDate: '2018-10-26 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
