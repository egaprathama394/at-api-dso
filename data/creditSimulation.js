export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};