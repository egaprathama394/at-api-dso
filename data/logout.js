export default {
  applicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '8B55A4DA-5331-47F9-9C8D-838A110FA18C',
};


export const negativeRequest = {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
