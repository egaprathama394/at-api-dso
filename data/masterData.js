
export default {
  companyCode: '0003',
  masterDataInfo: [
    {
      lastSync: '1900-01-01 00:00:00',
      objectName: 'Branch',
    },
  ],
  vehicleBrandCode: 'DAIHATSU',
  accountType: '1',
  action: '',
  deviceType: '0',
  operatingSystem: '0',
  serVersion: '1.0.0',
};

export const JobServiceCompany = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'JobService_Company',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const VehicleBrand = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'vehicleBrand',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const district = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'district',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const province = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'province',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const region = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'region',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const holidayCalendar = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'holidayCalendar',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

// looping data for job service company

export const loopVehicleType = [
  {
    vehicleType: 'AYLA',
    title: 'test case get master data job service company where vehicle = AYLA',
  },
  {
    vehicleType: 'GRANMAX',
    title: 'test case get master data job service company where vehicle = GRANMAX',
  },
  {
    vehicleType: 'GRANMAX1.3',
    title: 'test case get master data job service company where vehicle = GRANMAX1.3',
  },
  {
    vehicleType: 'GRANMAX1.5',
    title: 'test case get master data job service company where vehicle = GRANMAX1.5',
  },
  {
    vehicleType: 'LUXIO',
    title: 'test case get master data job service company where vehicle = LUXIO',
  },
  {
    vehicleType: 'Sigra 1.0',
    title: 'test case get master data job service company where vehicle = Sigra 1.0',
  },
  {
    vehicleType: 'Sigra 1.2',
    title: 'test case get master data job service company where vehicle = Sigra 1.2',
  },
  {
    vehicleType: 'SIRION',
    title: 'test case get master data job service company where vehicle = SIRION',
  },
  {
    vehicleType: 'TERIOS',
    title: 'test case get master data job service company where vehicle = TERIOS',
  },
  {
    vehicleType: 'XENIA1.0',
    title: 'test case get master data job service company where vehicle = XENIA1.0',
  },
  {
    vehicleType: 'XENIA1.3',
    title: 'test case get master data job service company where vehicle = XENIA1.3',
  },
  {
    vehicleType: 'OTHERS',
    title: 'test case get master data job service company where vehicle = OTHERS',
  },
];

export const VehicleTestDrive = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'VehicleTestDrive',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const parameterConfigApp = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'ParameterConfigApplication',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
