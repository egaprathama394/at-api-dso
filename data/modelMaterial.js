export default {
  AccountType: '1',
  Action: '',
  ApplicationName: '78A8D65B-666D-4FE1-B74C-15623C930C91',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0',
};
