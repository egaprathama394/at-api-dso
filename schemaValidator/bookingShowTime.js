export default {
  type: 'object',
  required: [
    'slot',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    slot: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'timeStart',
          'timeEnd',
        ],
        properties: {
          timeStart: {
            type: 'string',
          },
          timeEnd: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};
