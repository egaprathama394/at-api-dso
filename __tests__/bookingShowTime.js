import getTime from '../endpoints/bookingShowTime';
import * as TimeSchema from '../schemaValidator/bookingShowTime';
import * as body from '../data/bookingShowTime';

const fs = require('fs');
const { matchers } = require('jest-json-schema');

expect.extend(matchers);

const ambilToken = fs.readFileSync('./token.json');
const objToken = JSON.parse(ambilToken);

const token = objToken.edit;

describe('Display Booking Show Time', () => {
  test('Can display booking show time successfully using authorized token - positive test cases', async () => {
    const response = await getTime(token, body.default);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(TimeSchema.default);
    expect(response.body).toMatchSnapshot();
  });
  test('Can not display booking show time with invalid input date  - negative test cases', async () => {
    const response = await getTime(token, body.negativeRequest);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(TimeSchema.default);
    expect(response.body).toMatchSnapshot();
  });
});
