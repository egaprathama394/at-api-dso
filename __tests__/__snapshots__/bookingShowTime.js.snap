// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Booking Show Time Can display booking show time successfully using authorized token - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "build": "0",
  "slot": Array [
    Object {
      "timeEnd": "2020-09-30 09:00:00",
      "timeStart": "2020-09-30 08:00:00",
    },
    Object {
      "timeEnd": "2020-09-30 09:30:00",
      "timeStart": "2020-09-30 08:30:00",
    },
    Object {
      "timeEnd": "2020-09-30 10:00:00",
      "timeStart": "2020-09-30 09:00:00",
    },
    Object {
      "timeEnd": "2020-09-30 10:30:00",
      "timeStart": "2020-09-30 09:30:00",
    },
    Object {
      "timeEnd": "2020-09-30 11:00:00",
      "timeStart": "2020-09-30 10:00:00",
    },
    Object {
      "timeEnd": "2020-09-30 11:30:00",
      "timeStart": "2020-09-30 10:30:00",
    },
    Object {
      "timeEnd": "2020-09-30 12:00:00",
      "timeStart": "2020-09-30 11:00:00",
    },
    Object {
      "timeEnd": "2020-09-30 13:30:00",
      "timeStart": "2020-09-30 11:30:00",
    },
    Object {
      "timeEnd": "2020-09-30 14:00:00",
      "timeStart": "2020-09-30 13:00:00",
    },
    Object {
      "timeEnd": "2020-09-30 14:30:00",
      "timeStart": "2020-09-30 13:30:00",
    },
    Object {
      "timeEnd": "2020-09-30 15:00:00",
      "timeStart": "2020-09-30 14:00:00",
    },
    Object {
      "timeEnd": "2020-09-30 15:30:00",
      "timeStart": "2020-09-30 14:30:00",
    },
    Object {
      "timeEnd": "2020-09-30 16:00:00",
      "timeStart": "2020-09-30 15:00:00",
    },
  ],
  "version": "1.0",
}
`;

exports[`Display Booking Show Time Can not display booking show time with invalid input date  - negative test cases 1`] = `
Object {
  "acknowledge": 0,
  "build": "0",
  "message": "Waktu Booking tidak valid
",
  "slot": Array [],
  "version": "1.0",
}
`;
